crypto/fipsmodule/ec/internal.h

```diff
@@ -745,6 +745,46 @@ const EC_METHOD *EC_GFp_nistp521_method(void);
 // x86-64 optimized P256. See http://eprint.iacr.org/2013/816.
 const EC_METHOD *EC_GFp_nistz256_method(void);
 
+
+// EC_KEY METHOD
+
+// ec_key_method_st is a structure of function pointers implementing EC_KEY
+// operations. Currently, AWS-LC only allows consumers to use the |init|,
+// |finish|, |sign|, |sign_sig|, and |flag| fields.
+//
+// We do not support the |verify|, |verify_sig|, |compute_key|, or |keygen|
+// fields at all. If this struct is made public in the future, to maintain
+// OpenSSL compatability and match the struct size, they should be added in.
+struct ec_key_method_st {
+    int (*init)(EC_KEY *key);
+    void (*finish)(EC_KEY *key);
+
+    // AWS-LC doesn't support custom values for EC_KEY operations
+    // as of now. |k_inv| and |r| must be NULL parameters.
+    // The |type| parameter is ignored in OpenSSL, we pass in zero for it
+    int (*sign)(int type, const uint8_t *digest, int digest_len,
+                uint8_t *sig, unsigned int *siglen, const BIGNUM *k_inv,
+                const BIGNUM *r, EC_KEY *eckey);
+
+    // AWS-LC doesn't support custom values for EC_KEY operations
+    // as of now. |k_inv| and |r| must be NULL parameters.
+    ECDSA_SIG *(*sign_sig)(const uint8_t *digest, int digest_len,
+                           const BIGNUM *in_kinv, const BIGNUM *in_r,
+                           EC_KEY *eckey);
+
+    // Currently, |EC_KEY_METHOD| only supports |ECDSA_FLAG_OPAQUE|. It is
+    // not set by default.
+    int flags;
+
+    // AWS-LC currently does not support these fields, must be NULL
+    int (*sign_setup)(EC_KEY *eckey, BN_CTX *ctx_in, BIGNUM **k_inv,
+                      BIGNUM **r);
+    int (*copy)(EC_KEY *dest, const EC_KEY *src);
+    int (*set_group)(EC_KEY *key, const EC_GROUP *group);
+    int (*set_private)(EC_KEY *key, const BIGNUM *priv_key);
+    int (*set_public)(EC_KEY *key, const EC_POINT *pub_key);
```

Comment:
This is an opaque struct, so I don't see a reason to keep these field here for OpenSSL compatibility. If you don't think they should be deleted, could we comment them out?

Reply:
Got it, I just think it's good to have them since we offer setters for this functionality (although we assert people can't set these fields). I think commenting it out is a good compromise.